File: Greedy.dart
void coinChange(int amount) 
{
  List<int> coins = [2,1,5,10]; 
  // Sort in descending order
  coins.sort((b, a) => a.compareTo(b));

  for (int coin in coins) 
  {
    int count = amount ~/ coin;
    amount = amount - (coin * count); 
    if (count > 0) 
    {
      print("$coin x $count");
    }
  }
}

void main() 
{
  int amount = 87;
  coinChange(amount);
}

File: Greedy.cpp
#include <iostream>
using namespace std;

int coin[] = {10, 5, 2, 1};
int return_change(int amount)
 {
    if (amount < 0) return -1;

    int res = 0;
    int j = 0;
    int size = sizeof(coin) /sizeof(coin[0]);

    while (amount > 0) 
    {
        if (j >= size) return -1;
        if (amount >= coin[j]) 
        {
            amount -= coin[j];
            res++;
        }
        else
        {
            j++;
        }
    }
    return res;
}

int main() 
{
    int amount;
    cout << "Enter the amount: ";
    cin >> amount;
    int result = return_change(amount);
    if (result == -1) 
    {
        cout << "The amount cannot be represented" << endl;
    } 
    else 
    {
        cout << "Minimum coins required: " << result << endl;
    }
    return 0;
}
